src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -447,40 +474,47 @@ class FrontendGenImpl {
         inVals[inputIdx] = constantValue;
       }
     }
-
+    //
     assert(inVals[1] != nullptr && "Slice requires a starts attribute");
     assert(inVals[2] != nullptr && "Slice requires an ends attribute");
-    const auto startsType = inVals[1].getType().dyn_cast<RankedTensorType>();
-    assert(startsType != nullptr && "starts type is not a RankedTensorType");
-    auto startsDim = startsType.getShape()[0];
-
-    // If axes is not specified, default to [0, ..., ndim-1]
-    if (inVals[3] == nullptr) {
-      SmallVector<int64_t, 1> vals = {};
-      for (size_t s = 0; s < startsDim; ++s)
-        vals.emplace_back(s);
-      auto constantDenseAttribute =
-          mlir::DenseElementsAttr::get(tensorType, llvm::makeArrayRef(vals));
-      auto constantOp = builder_.create<mlir::ONNXConstantOp>(
-          UnknownLoc(), mlir::Attribute(), constantDenseAttribute);
-      mlir::Value constantResult = constantOp.output();
-      inVals[3] = constantResult;
-    }
-
-    // If steps is not specified, default to [1, ..., 1]
-    if (inVals[4] == nullptr) {
-      SmallVector<int64_t, 1> vals(startsDim, 1);
-      auto constantDenseAttribute =
-          mlir::DenseElementsAttr::get(tensorType, llvm::makeArrayRef(vals));
-      auto constantOp = builder_.create<mlir::ONNXConstantOp>(
-          UnknownLoc(), mlir::Attribute(), constantDenseAttribute);
-      mlir::Value constantResult = constantOp.output();
-      inVals[4] = constantResult;
-    }
+    inVals[3] = none();
```

Comment:
Where did the previous code go? Is the default value for none input no longer needed?

Reply:
I moved it under shape inference. I don't think any shape-aware computation should happen at the time we import onnx graph into ONNX Dialect.